common/network-common/src/main/java/org/apache/spark/network/crypto/README.md

```diff
@@ -99,3 +114,13 @@ sessions. It would, however, allow impersonation of future sessions.
 In the event of a pre-shared key compromise, messages would still be confidential from a passive
 observer. Only active adversaries spoofing a session would be able to recover plaintext.
 
+Security Changes & Compatibility
+-------------
+
+The original version of this protocol, retroactively called v1.0, did not apply an HKDF to `sharedSecret` and was
+directly using the encoded X coordinate as key material. This is atypical and standard practice is to pass that shared
+coordinate through an HKDF. The current version, v1.1, adds this additional HKDF to
+derive `derivedKey`.
```

Comment:
Essentially, this is backwardly incompatible with 3.x  on the wire - making adoption of 4.x in secure env all-or-nothing
Given this, I am leaning towards flag guarding the change.

Reply:
+1 for @mridulm 's recommendation.